signingscript/utils.py

```diff
@@ -156,3 +159,58 @@ def copy_to_dir(source, parent_dir, target=None):
 
     if exitcode != 0:
         raise FailedSubprocess('Command `{}` failed'.format(' '.join(command)))
+
+
+async def sign_file_with_autograph(context, from_, fmt, to=None):
+    """Signs a file with autograph and writes the result to arg
+    `to` or `from_` if `to` is None.
+
+    Args:
+        context (Context): the signing context
+        from_ (str): the source file to sign
+        fmt (str): the format to sign with
+        to (str, optional): the target path to sign to. If None, overwrite
+            `from_`. Defaults to None.
+
+    Raises:
+        Requests.RequestException: on failure
+    """
+    to = to or from_
+
+    # build and run the signature request
+    input_bytes = open(from_, 'rb').read()
+    sign_req = [{
+        "input": base64.b64encode(input_bytes)
+    }]
+    log.debug("autograph signature request: %s" % sign_req)
+    base_url, auth = get_autograph_url_and_auth(context.config)
+    url = "%s/sign/file" % base_url
+    loop = asyncio.get_event_loop()
+    make_sign_req = functools.partial(requests.post, url, json=sign_req, auth=auth)
+    response = await loop.run_in_executor(None, make_sign_req)
```

Comment:
TIL. Is there a reason we're using `requests.post` this way instead of `aiohttp.request('post', ...)` or `session.post()` where session is an aiohttp session? Because of `requests_hawk`?

+1 in adding timeouts and retries. There's a [`retry_request`](https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/utils.py#L83) and [`retry_async`](https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/utils.py#L229) in scriptworker you're welcome to take advantage of. If we define the exceptions that we want to retry on, and either pass a `timeout` kwarg to requests or [aiohttp](https://docs.aiohttp.org/en/stable/client_quickstart.html?highlight=timeout#timeouts), we could have `n` retries with exponential backoff.